helix-view/src/graphics.rs

```diff
@@ -170,7 +178,8 @@ impl Rect {
     }
 
     /// Calculate the intersection between two [`Rect`]s.
-    pub fn intersection(self, other: Rect) -> Rect {
+    #[must_use]
+    pub fn intersection(self, other: Self) -> Self {

```

Comment:
Is this a clippy suggestion? I personally prefer the older version because it's clearer.

Reply:
I prefer Self but this looks like an optional clippy lint that's off by default